### PR TITLE
fix(AIP-158): reference resource plural for field name

### PR DESCRIPTION
**Pull request by noahdietz**
As we start to require `plural` in `google.api.resource`, we should referencing it in rules that attempt to pluralize words for things like field names or collection IDs. This is a reasonable heuristic to use for the AIP-158 `response-plural-first-field` when that first field is a resource.

Fixes http://b/271479822

### CODE REVIEW

File: rules/aip0158/response_plural_first_field.go [hunk 1]
```diff
@@ -30,7 +31,16 @@ var responsePluralFirstField = &lint.MessageRule{
...
 		firstField := m.GetFields()[0]
-		want := utils.ToPlural(firstField.GetName())
+
+		// If the field is a resource, use the `plural` annotation to decide the
+		// appropriate plural field name.
+		var want string
+		if res := utils.GetResource(firstField.GetMessageType()); res != nil && res.GetPlural() != "" {
+			want = strcase.SnakeCase(res.GetPlural())
+		} else {
+			want = utils.ToPlural(firstField.GetName())
```
Comment (toumorokoshi) [inline rules/aip0158/response_plural_first_field.go:41]: this already existed of course, but I feel like pluralizing the resource singular or the type would be more robust than the field name.

File: rules/aip0158/response_plural_first_field.go [hunk 2]
```diff
@@ -30,7 +31,16 @@ var responsePluralFirstField = &lint.MessageRule{
 		// Throw a linter warning if, the first field in the message is not named
 		// according to plural(message_name.to_snake().split('_')[1:-1]).
 		firstField := m.GetFields()[0]
-		want := utils.ToPlural(firstField.GetName())
+
+		// If the field is a resource, use the `plural` annotation to decide the
+		// appropriate plural field name.
+		var want string
+		if res := utils.GetResource(firstField.GetMessageType()); res != nil && res.GetPlural() != "" {
```
Comment (toumorokoshi) [inline rules/aip0158/response_plural_first_field.go:38]: nit: make a util function similar to GetResourceSingular? https://github.com/googleapis/api-linter/blob/ececf6c459201eccbaf506e57d0f735c62ba7aa0/rules/internal/utils/resource.go#L25.

**Review by toumorokoshi — APPROVED** [review]

File: rules/aip0158/response_plural_first_field.go (hunk 1 above)
Comment (noahdietz) [reply rules/aip0158/response_plural_first_field.go:41]: I think this is meant to support the scenario where the paginated field isn't a resource. I agree with you, but I don't know if we should change it right now. There are maybe one or two Standard List-like fields that paginate a `repeated string` field that I know of. This rule being in AIP-158, not AIP-132 is poignant.

File: rules/aip0158/response_plural_first_field.go (hunk 2 above)
Comment (noahdietz) [reply rules/aip0158/response_plural_first_field.go]: Good call! Done.

File: rules/aip0158/response_plural_first_field.go (hunk 1 above)
Comment (noahdietz) [reply rules/aip0158/response_plural_first_field.go:41]: Opened #1160

**Pull request merged**